docs/concepts/app-anatomy.md

```diff
@@ -0,0 +1,214 @@
+
+To learn more about the `start` command, [click here](./node.md#start-command).
+
+## Core Application File
+
+In general, the core of the state-machine is defined in a file called `app.go`. It mainly contains the **type definition of the application** and functions to **create and initialize it**. 
+
+### Type Definition of the Application
+
+The first thing defined in `app.go` is the `type` of the application. It is generally comprised of the following parts:
+
+- **A reference to [`baseapp`](./baseapp.md).** The custom application defined in `app.go` is a Golang embedding of the `baseapp` type. `baseapp` implements most of the core logic for the application, including all the [ABCI methods](https://tendermint.com/docs/spec/abci/abci.html#overview) and the routing logic. When a transaction is relayed by Tendermint to the application, the latter uses `baseapp`'s methods to route them to the appropriate module. 
+- **A list of store keys**. The [store](./store.md), which contains the entire state, is implemented as a multistore (i.e. a store of stores) in the Cosmos SDK. Each module uses one or multiple stores in the multistore to persist their part of the state. These stores can be accessed with specific keys that are declared in the `app` type. These keys, along with the `keepers`, are at the heart of the [object-capabilities model](../intro/ocap.md) of the Cosmos SDK.  
+- **A list of module's `keepers`.** Each module defines an abstraction called `keeper`, which handles reads and writes for this module's store(s). The `keeper`'s methods of one module can be called from other modules (if authorized), which is why they are declared in the application's type. 
+- **A reference to a `codec`.** The Cosmos SDK gives developers the freedom to choose the encoding framework for their application. The application's `codec` is used to serialize and deserialize data structures in order to store them, as stores can only persist `[]bytes`. The `codec` must be deterministic. Most SDK application use [amino](./amino.md) as their `codec`. 
+
+You can see an example of application type definition [here](https://github.com/cosmos/sdk-application-tutorial/blob/master/app.go#L27-L43).
+
+### Constructor Function
+
+This function constructs a new application of the type defined above. It is [called](https://github.com/cosmos/cosmos-sdk/blob/master/server/start.go#L117) every time the full-node is started with the `start` command. Here are the main actions performed by this function:
+
+- Instanciate a new application with a reference to a `baseapp` instance, a codec and all the appropriate store keys.
+- Instantiate all the `keepers` defined in the application's `type`.
```

Comment:
might be good to add a point (or subpoint) regarding all the specialized module functionality which is defined with a new app, such as setting invariants (crisis) or adding custom proposal handlers client functionality (gov) - all the things which tie the keepers together

Reply:
++

Reply:
Yes, good point. Actually I am going to rework the doc a bit to include the new logic of the module manager